### DISCUSSION TITLE
Sparse matrices in skrf?

### DISCUSSION THREAD

**Discussion by fffrune** [scikit-rf/scikit-rf#794] [Ideas]
Is skrf able to leverage sparse matrices?  I realize that this is not a typical use case in RF engineering, but it's not unthinkable either.  For example, I'm looking at arrays of coupled stubs and beyond the first few neighbors, the coupling is negligible.  So instead of carrying around a large NxN S-matrix, I'd rather just have 3xN or 5xN data points.  Has this been considered?  Would it be a big job?  I'm new to skrf so please forgive my ignorance.

**Comment by jhillairet** [comment]
Hi,

I don't think this has ever been considered. You're right that in some case
some of the matrices should be sparse.

I'm not sure however where and how sparse matrices should be implemented
instead of standard numpy arrays. Networks objects implement s parameters
as a (nb_freq, nb_s, nb_s) numpy arrays. Do you already have a precise idea?
